### PR TITLE
Move arming checks for terrain-all-loaded up to common Arming library

### PR DESCRIPTION
**Pull request by peterbarker**
Both tridge and I thought the "all loaded" stuff was present for Plane....

### CODE REVIEW

File: ArduCopter/AP_Arming.cpp [hunk 1]
```diff
@@ -116,6 +116,17 @@ bool AP_Arming_Copter::board_voltage_checks(bool display_failure)
     return true;
 }
 
+// expected to return true if the terrain database is healthy and able
```
Comment (peterbarker) [inline ArduCopter/AP_Arming.cpp:119]: Bad comment

File: ArduCopter/AP_Arming.h [hunk 2]
```diff
@@ -51,6 +51,10 @@ class AP_Arming_Copter : public AP_Arming
 
     void set_pre_arm_check(bool b);
 
+    // expected to return true if the terrain database is healthy and able
```
Comment (peterbarker) [inline ArduCopter/AP_Arming.h:54]: Bad comment

**Comment by tridge** [comment]
thanks! important to get this fixed

**Comment by peterbarker** [comment]
This is now doing a lot more than making the code common.  Several bugs are being fixed in our terrain prearm checks.

Some of the new checks may be redundant (it may not be possible to both have an empty cache and have the have-terrain flag be true).

**Review by tridge — APPROVED** [review]

**Review by davidbuzz — APPROVED** [review]

File: libraries/AP_Mission/AP_Mission.h [hunk 3]
```diff
@@ -619,6 +619,9 @@ class AP_Mission
     // returns true if the mission contains the requested items
     bool contains_item(MAV_CMD command) const;
 
+    // returns true if the mission has a terrain relative mission item
+    bool contains_terrain_relative(void);
```
Comment (rmackay9) [inline libraries/AP_Mission/AP_Mission.h:623]: can we rename "constains_terrain_relative" to "contains_terrain_alt"?  No big deal of course but "terrain_alt" is what we use in Copter so it's more likely to show up in searches.

Comment (peterbarker) [reply libraries/AP_Mission/AP_Mission.h:623]: Created a PR for that name change.

**Pull request merged**